pkg/windows/calico.go

```diff
@@ -60,25 +62,47 @@ func (c *Calico) Setup(ctx context.Context, dataDir string, nodeConfig *daemonco
 		return nil, err
 	}
 
-	secrets, err := client.CoreV1().Secrets(CalicoSystemNamespace).List(ctx, metav1.ListOptions{})
+	sa, err := serviceaccount.GetOrCreateServiceAccount(client.CoreV1(), CalicoSystemNamespace, calicoNode)
 	if err != nil {
 		return nil, err
 	}
 
-	for _, secret := range secrets.Items {
-		if !isCalicoNodeToken(&secret) {
-			continue
-		}
-		value, ok := secret.Data["token"]
-		if !ok {
-			continue
-		}
-		calicoKubeConfig.Token = string(value)
+	t := true
+	sa.AutomountServiceAccountToken = &t
+
+	saClient := client.CoreV1().ServiceAccounts(CalicoSystemNamespace)
+
+	req := authv1.TokenRequest{
+		TypeMeta:   sa.TypeMeta,
+		ObjectMeta: sa.ObjectMeta,
+		Spec: authv1.TokenRequestSpec{
+			BoundObjectRef: &authv1.BoundObjectReference{
+				Kind:       sa.Kind,
+				APIVersion: sa.APIVersion,
+				Name:       sa.Name,
+			},
+		},
 	}
-	if calicoKubeConfig.Token == "" {
-		return nil, errors.New("could not retrieve calico node token from the cluster")
+
+	token, err := saClient.CreateToken(ctx, sa.Name, &req, metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+	tokenStatus := token.Status
+	if tokenStatus.Token == "" {
+		return nil, fmt.Errorf("[Setup] calico-node service account (%s) returned an empty token (%s) from the cluster", sa.Name, token.Name)
+	}
```

Comment:
If the request is successful, you're guaranteed to get a Token back. You shouldn't need to check for both `err != nil` and `Token == ""`

```suggestion
	if err != nil {
		return nil, errors.Wrapf(err, "failed to create token for service account %s/%s", CalicoSystemNamespace, calicoNode)
	}
```